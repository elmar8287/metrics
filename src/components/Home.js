import { React, useState, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { fetchCountriesFromApi } from '../redux/home';
import Data from './Data';
import Navbar from './Navbar';
import Country from './Country';
import WorldData from './WorldData';
import '../assets/css/style.css';

const Home = () => {
  const state = useSelector((state) => state.countriesReducer);
  const { oneCountry, totalConfirmed } = state;

  const dispatch = useDispatch();

  const [filter, setFilter] = useState('');

  useEffect(() => {
    dispatch(fetchCountriesFromApi());
  }, []);

  const onChange = (e) => {
    setFilter(e.target.value);
  };

  return (
    <>
      <Navbar text="WorldData Covid-19 Stats" year="2021" />
      <WorldData totalConfirmed={totalConfirmed} />
      <section>
        <Data filter={filter} onChange={onChange} />
        <h3 className="stats-name">Stats By Country</h3>
        <ul className="container">
          {oneCountry && oneCountry
            .filter((location) => location.name.toLowerCase().includes(filter.toLowerCase()) || filter === '')
            .map((country) => {
              const {
                name,
                confirmedCases,
                id,
                allDeaths,
                allRegions,
                recovered,
                openCases,
              } = country;
              return (
                <Link
                  className="country-card"
                  key={id}
                  to={{
                    pathname: `/details/${name}`,
                    state: {
                      countryName: name,
                      countryId: id,
                      casesConfirmed: confirmedCases,
                      deaths: allDeaths,
                      regions: allRegions,
                      allRecovered: recovered,
                      casesOpen: openCases,
                    },
                  }}
                >
                  <Country name={name} confirmedCases={confirmedCases} />
                </Link>
              );
            })}
        </ul>
      </section>
    </>
  );
};

export default Home;
