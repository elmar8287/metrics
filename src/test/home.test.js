import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configStore';
import Home from '../components/Home';

describe('Home page render correctly', () => {
  test('Home component renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Provider store={store}>
            <Home />
          </Provider>
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
