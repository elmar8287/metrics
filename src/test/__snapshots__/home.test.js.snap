// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page render correctly Home component renders correctly 1`] = `
Array [
  <section>
    <div>
      <a
        href="/"
        onClick={[Function]}
      >
        <p>
          Home page
        </p>
      </a>
      <h1>
        2021
      </h1>
    </div>
    <div>
      <h2>
        World Covid-19 Stats
      </h2>
    </div>
    <div>
      <p>
        something
      </p>
    </div>
  </section>,
  <section>
    <div>
      <div>
        <p>
          icon maybe
        </p>
      </div>
      <aside>
        <h2>
          World
        </h2>
        <span>
          0 cases
        </span>
      </aside>
    </div>
  </section>,
  <section>
    <div>
      <input
        onChange={[Function]}
        placeholder="Search Country"
        type="text"
        value=""
      />
    </div>
    <h3>
      Stats By Country
    </h3>
    <ul />
  </section>,
]
`;
